src/Workspaces/Core/Portable/Workspace/Host/HostProjectServices.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    // TODO(cyrusn): Make public.  Tracked through https://github.com/dotnet/roslyn/issues/62914
+    internal sealed class HostProjectServices
```

Comment:
>class [](http://example.com/codeflow?start=20&length=5)

Can/should this be a struct?

Reply:
i'm very very hesitant on structs as it is a breaking change to make them classes and it's hard to tell up front if being a struct will be an issue for us (for example, having to deal with potentially default values).

Reply:
default value is the same as null in this case, in public APIs we would need to check either way